transport/server.go

```
@@ -202,13 +207,21 @@ func (s *server) listenTCP() error {
+		// previously a nil config with nil err fell through, leaving
+		// streamListener nil, and s.streamListener.Addr() panicked below.
+		sslConfig, buildTlsConfErr := s.tlsConfigBuilder.BuildTlsConfig()
+		if buildTlsConfErr != nil {
+			return perrors.Wrapf(buildTlsConfErr, "BuildTlsConfig(addr:%s)", s.addr)
+		}
+		if sslConfig == nil {
+			return fmt.Errorf("BuildTlsConfig returned nil config without error for addr:%s", s.addr)
 		}
+		streamListener, err = tls.Listen("tcp", s.addr, sslConfig)
+	} else {
+		streamListener, err = net.Listen("tcp", s.addr)
```

golangci-lint (2.12.2)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)